bip-0008.mediawiki

```diff
@@ -93,7 +95,9 @@ During the MUST_SIGNAL phase, if '''(2016 - threshold)''' blocks in the retarget
 
 <img src="bip-0008/states.png" align="middle"></img>
 
-Note that when '''lockinontimeout''' is true, the LOCKED_IN state will be reached no later than at a height of '''timeoutheight''', and ACTIVE will be reached no later than at a height of '''timeoutheight + 2016'''.
+Note that when '''lockinontimeout''' is true, the LOCKED_IN state will be reached no later than at a height of '''timeoutheight'''.
+Regardless of value of '''lockinontimeout''', ACTIVE will be reached no earlier than at a height of '''minimum_activation_height'''.
+When '''minimum_activation_height''' is less than or equal to '''timeoutheight + 2016''', ACTIVE will be reached no later than '''timeoutheight + 2016'''.
```

Comment:
nit: probably (slightly) easier to understand if this wasn't special cased and instead a note would be added to the parameter selection guidelines that `minimum_activation _height` must be at least `timeoutheight + 2016`. Then you could also remove the `After at least one retarget period of LOCKED_IN [...]` and change the sentence to something like "From the LOCKED_IN state we transition to ACTIVE if the minimum activation height is reached."

Reply:
Afaics right now the diagram isn't correct, because if `minimum_activation_height = 0` then it looks like LOCKED_IN -> ACTIVE happens without a retarget period. So should either make `minimum_activation_height >= timeoutheight + 2016` or change arrow in diagram to `height >= max(timeoutheight + 2016, minimum_activation_height)`.

Reply:
`minimum_activation_height` can't be at least `timeoutheight + 2016` as otherwise the normal deployment won't activate one retarget period after LOCKED_IN.

Reply:
Of course, nevermind. In that case the diagram isn't made worse by this PR.

Reply:
FWIW, I personally intended that a min-activation-height that is less than timeoutheight + 2016 would be allowed in general.  That said, my personal intentions may not be all that relevant.

Reply:
@jonasnick "it looks like LOCKED_IN -> ACTIVE happens without a retarget period" -- if that were going to happen, there'd be an arrow from "STARTED -> ACTIVE" without going through "LOCKED_IN" -- if you enter a state you're there for a retarget period, and you get to follow one of the arrows (maybe just the loop back and stay in the same state) when you hit the next retarget period.